dev/io.openliberty.session.6.1.internal/src/io/openliberty/session/AbstractSessionData61.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.session;
+
+import java.util.function.Consumer;
+import java.util.logging.Level;
+
+import com.ibm.ejs.ras.TraceNLS;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.session.AbstractSessionData;
+import com.ibm.ws.session.SessionContext;
+import com.ibm.ws.session.utils.LoggingUtil;
+import com.ibm.wsspi.session.ISession;
+
+import jakarta.servlet.ServletContext;
+import jakarta.servlet.http.HttpSession;
+
+/*
+ * Servlet 6.1 - support HttpSession getAccessor() API
+ * 
+ * @see jakarta.servlet.http.HttpSession#getAccessor()
+ * @see jakarta.servlet.http.HttpSession.Accessor#access(Consumer<HttpSession>)
+ */
+public abstract class AbstractSessionData61 extends AbstractSessionData {
+    private static final String CLASS_NAME = "AbstractSessionData61";
+    private static TraceNLS nls = TraceNLS.getTraceNLS(AbstractSessionData61.class, "io.openliberty.session.resources.SessionMessages");
+
+    protected AbstractSessionData61(ISession session) {
+        super(session);
+
+        if (TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
+            LoggingUtil.SESSION_LOGGER_CORE.log(Level.FINE, CLASS_NAME + " Constructor (ISession) [" + session + "]");
+        }
+    }
+
+    protected AbstractSessionData61(ISession session, SessionContext sessCtx, ServletContext servCtx) {
+        super(session, sessCtx, servCtx);
+
+        if (TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINER)) {
+            LoggingUtil.SESSION_LOGGER_CORE.log(Level.FINE, CLASS_NAME + " Constructor , ISession [ " + session + " ] ; ServletContext [" + servCtx + "]");
+        }
+    }
+
+    public class SessionDataAccessor implements HttpSession.Accessor {
+        private final String INNER_CLASS_NAME = "SessionDataAccessor";
+        private final ISession iSession;
+        String savedSessionId = null;
+
+        public SessionDataAccessor(ISession session, String id) {
+            iSession = session;
+            savedSessionId = id;
+
+            if (TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
+                LoggingUtil.SESSION_LOGGER_CORE.log(Level.FINE,
+                                                    INNER_CLASS_NAME + " Constructor , session id [" + savedSessionId + "] , this [" + this + "] . ISession [" + iSession + "]");
+            }
+        }
+
+        /**
+         * Wrap and create a HttpSession reference which application can access via the passed in {@code sessionConsumer}
+         * Update the last accessed time.
+         */
+        @Override
+        public void access(Consumer<HttpSession> sessionConsumer) {
+            if (TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINE)) {
```

Comment:
```suggestion
            if (TraceComponent.isAnyTracingEnabled() && LoggingUtil.SESSION_LOGGER_CORE.isLoggable(Level.FINER)) {
```
entering and existing are FINER log calls.